projects/cm_mcu/LocalTasks.c

```diff
@@ -659,95 +659,80 @@ void init_registers_ff()
-    }
-    if ((i+1) % 3 == 2){
-      reg_pages[1] = reg_page; // updated page of a low-byte address per a triplet
-    }
-    if ((i+1) % 3 == 0){ // this is when we retrieve two-byte address and data stored in three sequential lines from eeprom
-      reg_pages[2] = reg_page; // updated page of data per a triplet
-      uint16_t packed_reg0_address = (reg_pages[0] << 8) + (i-2)%128 ;
-      apollo_i2c_ctl_reg_r(2, 0x50, 2, packed_reg0_address, 1, &reg0); //read reg0 register from eeprom
-
-      if (reg_pages[0] == reg_pages[1] && reg_pages[0] == reg_pages[2]){
-        apollo_i2c_ctl_reg_r(2, 0x50, 2, packed_reg0_address, 3, &triplet); //read triplet from eeprom
-        // organize the three bytes
-        data = triplet >> 16 ;
-        reg1 = (triplet - (data << 16)) >> 8;
-        reg0 = triplet - (data << 16) - (reg1 << 8);
 
-      }
-      else{ // an edge case of a triplet not stored on the same eeprom page
-
-        uint16_t packed_reg1_address = (reg_pages[1] << 8) + (i-1)%128;
-        apollo_i2c_ctl_reg_r(2, 0x50, 2, packed_reg1_address, 1, &reg1); //read reg1 register from eeprom
-        uint16_t packed_data_address = (reg_pages[2] << 8) + i%128;
-        apollo_i2c_ctl_reg_r(2, 0x50, 2, packed_data_address, 1, &data); //read data register from eeprom
+    if ((i+1) % 3 == 0){ // this is when we retrieve two-byte address and data stored in three sequential lines from eeprom
 
+      uint16_t packed_reg0_address = (reg_page << 8) + (i-2)%126 ;
+      int status_r = apollo_i2c_ctl_reg_r(2, 0x50, 2, packed_reg0_address, 3, &triplet); //read triplet from eeprom
+      if (status_r!= 0) {
+        log_debug(LOG_SERVICE, "error in read triplet from EEPROM");
+        log_error(LOG_SERVICE, "read status is %d\r\n",status_r);
       }
-
+      // organize the three bytes
+      data = (triplet >> 16) & 0xFFU;
+      reg1 = (triplet >> 8) & 0xFFU;
+      reg0 = triplet & 0xFFU;
 
       if (reg0 != HighByte) {
         log_debug(LOG_SERVICE, "check page written to clk = %x\r\n", reg0);
-        status = apollo_i2c_ctl_reg_w(2, i2c_addrs, 1, 0x01, 1, reg0); // write a page change to a clock chip
+        status_w = apollo_i2c_ctl_reg_w(2, i2c_addrs, 1, 0x01, 1, reg0); // write a page change to a clock chip
+        if (status_w != 0){
+          log_debug(LOG_SERVICE, "error in write status for page");
+          log_error(LOG_SERVICE, "write status is %d\r\n",status_w);
+          return status_w; // fail writing and exit
+        }
       }
 
       HighByte = reg0; //update the current high byte or page
 
-      if (status != 0){
-        log_debug(LOG_SERVICE, "error in write status for page");
-        log_error(LOG_SERVICE, "write status is %d\r\n",status);
-        return status; // fail writing and exit
-      }
 
-      vTaskDelay(pdMS_TO_TICKS(30)); //300 ms minimum
+      vTaskDelay(pdMS_TO_TICKS(10)); //delay 10 ms to prevent PSMON stack overflow
```

Comment:
I don't understand how a 10 ms delay could prevent a stack overflow. Can you confirm that this is needed?

Reply:
it wasn't needed as the way it is in  https://github.com/apollo-lhc/cm_mcu/blob/read_eeprom/projects/cm_mcu/LocalTasks.c#L703 but the stack overflow happens once log_debug is changed to log_info. 

Here is the issue:
`20673 SRV INF LocalTasks.c:706:check data written to clk = 0
20675 SRV INF LocalTasks.c:706:check data written to clk = 0
20677 SRV INF LocalTasks.c:706:check data written to clk = 0
20677 SRV INF MonitorTask.c:265:stack (PSMON) = 1Stack overflow: task PSMON `

Reply:
But, yes -- the 10 ms can prevent this type of overflow.

Reply:
increased stack size on PSMON once this was understood.